File: code/JL_OTA_V1.7.1_SDK_V1.9.2/otasdk/src/main/java/com/jieli/otasdk/activities/ContentActivity.java
package com.jieli.otasdk.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import com.jieli.component.utils.SystemUtil;
import com.jieli.otasdk.R;
import com.jieli.otasdk.base.BaseActivity;
import com.jieli.otasdk.databinding.ActivityContentBinding;

import java.util.HashMap;
import java.util.Map;

public class ContentActivity extends BaseActivity {
    public static final String FRAGMENT_TAG = "fragment_tag";
    public static final String FRAGMENT_DATA = "fragment_data";

    private static final long MIN_START_SPACE_TIME = 1000;//两次内容页面打开的最小时间间隔
    private static final Map<String, Long> fastClickLimit = new HashMap<>();

    private ActivityContentBinding mContentBinding;


    public static void startContentActivity(Context context, String fragmentCanonicalName) {
        startContentActivity(context, fragmentCanonicalName, null);
    }

    public static void startContentActivity(Context context, String fragmentCanonicalName, Bundle bundle) {
        if (null == context || fastStart(fragmentCanonicalName)) return;
        Intent intent = new Intent(context, ContentActivity.class);
        intent.putExtra(ContentActivity.FRAGMENT_TAG, fragmentCanonicalName);
        intent.putExtra(ContentActivity.FRAGMENT_DATA, bundle);
        context.startActivity(intent);
    }

    public static void startContentActivityForResult(Fragment fragment, String fragmentCanonicalName, Bundle bundle, int code) {
        if (null == fragment || fastStart(fragmentCanonicalName)) return;
        Intent intent = new Intent(fragment.requireContext(), ContentActivity.class);
        intent.putExtra(ContentActivity.FRAGMENT_TAG, fragmentCanonicalName);
        intent.putExtra(ContentActivity.FRAGMENT_DATA, bundle);
        fragment.startActivityForResult(intent, code);
    }

    //检测两次的打开时间间隔，判断是否重复打开页面
    private static boolean fastStart(String fragmentName) {
        if (null == fragmentName) return true;  //错误参数，直接不予处理
        Long startTime = fastClickLimit.get(fragmentName);
        long currentStartTime = System.currentTimeMillis();
        if (startTime != null && currentStartTime - startTime < MIN_START_SPACE_TIME) {  //判断属于快速点击，不予处理
            return true;
        }
        fastClickLimit.put(fragmentName, currentStartTime);
        return false;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SystemUtil.setImmersiveStateBar(getWindow(), true);
        mContentBinding = ActivityContentBinding.inflate(getLayoutInflater());
        setContentView(mContentBinding.getRoot());
        String fragmentName = getIntent().getStringExtra(FRAGMENT_TAG);
        if (null == fragmentName) {
            finish();
            return;
        }
        Bundle bundle = getIntent().getParcelableExtra(FRAGMENT_DATA);
        replaceFragment(R.id.cl_content, fragmentName, bundle);
    }


    @Override
    protected void onDestroy() {
        super.onDestroy();
        fastClickLimit.clear();
    }
}

File: code/JL_OTA_Android_V1.9.0_SDK_V1.11.0/otasdk/src/main/java/com/jieli/otasdk/ui/base/ContentActivity.kt
package com.jieli.otasdk.ui.base;

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.activity.result.ActivityResultLauncher
import com.jieli.component.utils.SystemUtil
import com.jieli.otasdk.R
import com.jieli.otasdk.databinding.ActivityContentBinding

/**
 * @author zqjasonZhong
 * @since 2025/1/13
 * @email zhongzhuocheng@zh-jieli.com
 * @desc 通用界面
 */
class ContentActivity : BaseActivity() {
    companion object{
        const val KEY_CLASS_PATH = "class_path"
        const val KEY_BUNDLE = "bundle"

        private const val MIN_START_ACTIVITY_INTERVAL = 500L
        private val fastClickLimit = HashMap<String, Long>()

        fun startContentActivity(
            context: Context,
            fragmentClazzPath: String?,
            bundle: Bundle? = null,
            launcher: ActivityResultLauncher<Intent>? = null
        ) {
            if (null == fragmentClazzPath) return
            if (fastStart(fragmentClazzPath)) return
            val intent = Intent(context, ContentActivity::class.java).also {
                it.putExtra(KEY_CLASS_PATH, fragmentClazzPath)
                bundle?.let { data ->
                    it.putExtra(KEY_BUNDLE, data)
                }
            }
            if (null == launcher) {
                context.startActivity(intent)
            } else {
                launcher.launch(intent)
            }
        }

        private fun fastStart(fragmentClazzPath: String): Boolean {
            val startTime = fastClickLimit[fragmentClazzPath]
            val currentTime = System.currentTimeMillis()
            if (null == startTime) {
                fastClickLimit[fragmentClazzPath] = currentTime
                return false
            }
            if (currentTime - startTime < MIN_START_ACTIVITY_INTERVAL) {
                return true
            }
            fastClickLimit[fragmentClazzPath] = currentTime
            return false
        }
    }

    private lateinit var binding: ActivityContentBinding
    private var fragmentClazzPath: String? = null
    var customExit: (() -> Boolean)? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        SystemUtil.setImmersiveStateBar(window, true)
        super.onCreate(savedInstanceState)
        binding = ActivityContentBinding.inflate(layoutInflater)
        setContentView(binding.root);
        val fragmentClazzPath = intent?.getStringExtra(KEY_CLASS_PATH)
        if (null == fragmentClazzPath) {
            finish()
            return
        }
        val bundle = intent?.getBundleExtra(KEY_BUNDLE)
        this.fragmentClazzPath = fragmentClazzPath
        replaceFragment(R.id.main, fragmentClazzPath, bundle, true)
    }

    override fun onDestroy() {
        super.onDestroy()
        fragmentClazzPath?.let {
            fastClickLimit.remove(it)
        }
    }

    @Deprecated("Deprecated in Java")
    @Suppress("DEPRECATION")
    override fun onBackPressed() {
        customExit?.let { method ->
            if (method()) {
                return
            }
        }
        super.onBackPressed()
    }
}